# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/CanvasCaptureMediaStream.h
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/CanvasCaptureMediaStream.h
--- a/dom/media/CanvasCaptureMediaStream.h	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/CanvasCaptureMediaStream.h	Wed Jul 31 07:58:17 2019 +0000
@@ -84,13 +84,12 @@
    */
   virtual void Forget() {}
 
- protected:
-  virtual ~OutputStreamDriver();
-
- private:
   const TrackID mTrackId;
   const RefPtr<SourceMediaStream> mSourceStream;
   const PrincipalHandle mPrincipalHandle;
+
+ protected:
+  virtual ~OutputStreamDriver();
 };
 
 class CanvasCaptureMediaStream : public DOMMediaStream {
@@ -102,7 +101,7 @@
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(CanvasCaptureMediaStream,
                                            DOMMediaStream)
 
-  nsresult Init(const dom::Optional<double>& aFPS, const TrackID& aTrackId,
+  nsresult Init(const dom::Optional<double>& aFPS, const TrackID aTrackId,
                 nsIPrincipal* aPrincipal);
 
   JSObject* WrapObject(JSContext* aCx,
@@ -119,12 +118,7 @@
    */
   void StopCapture();
 
-  /**
-   * Create a CanvasCaptureMediaStream whose underlying stream is a
-   * SourceMediaStream.
-   */
-  static already_AddRefed<CanvasCaptureMediaStream> CreateSourceStream(
-      nsPIDOMWindowInner* aWindow, HTMLCanvasElement* aCanvas);
+  SourceMediaStream* GetSourceStream() const;
 
  protected:
   ~CanvasCaptureMediaStream();